lib/krikri/mapper.rb

```diff
@@ -90,23 +90,55 @@ def mapping_exception_desc(rec)
     ##
     # A SoftwareAgent to run mapping processes.
     #
+    # @example
+    #
+    #   To map the records harvested by the harvest activity with ID 1:
+    #
+    #   Krikri::Mapper::Agent.enqueue(
+    #     :mapping,
+    #     opts = {
+    #       name: 'scdl_qdc',
+    #       generator_uri: 'http://ldp.local.dp.la/ldp/activity/1'
+    #     }
+    #   )
+    #   Or, supposing you already have a harvest Activity instance named
+    #   `my_activity':
+    #   [ ... ]
+    #     opts = {
+    #       name: 'scdl_qdc',
+    #       activity: my_activity
+    #     }
+    #
     # @see: Krikri::SoftwareAgent, Krikri::Activity
     class Agent
       include SoftwareAgent
 
-      attr_reader :name, :generator_uri
+      attr_reader :name
 
       def self.queue_name
         :mapping
       end
 
+      def self.entity_behavior
+        Krikri::AggregationEntityBehavior
+      end
+
+      def self.generator_entity_behavior
```

Comment:
This is a good point. The main difference seems to be that `queue_name` is used from the class method `.enqueue`. It can eventually be overridden at the method call level (https://github.com/dpla/KriKri/blob/develop/lib/krikri/software_agent.rb#L95).

> Why would we ever want to say that a mapper does not generate aggregations?

Mappers accept a model class on definition, with the idea of supporting conversion to non-DPLA MAP models: https://github.com/dpla/KriKri/blob/feature/index-an-activity/lib/krikri/mapper.rb#L41 

It might be neat to have the Agent infer its behavior (or maybe just the class to build) from the mapper itself.